apps/frontend/src/app/5_pages/MarketMakingPage/components/BobRepositionModal/hooks/useHandleSubmit.ts

```diff
@@ -0,0 +1,160 @@
+import { mintArgsForReposition } from '../BobRepositionModal.utils';
+
+const testAllowance = async (
+  owner: string,
+  token: CrocTokenView,
+  amount: BigNumber,
+) => {
+  const allowance = await token.allowance(owner);
+
+  if (allowance.lt(amount)) {
+    const approval = await token.approve();
+    return approval;
+  }
+};
+
+export const useHandleSubmit = (
+  assetA: string,
+  assetB: string,
+  position: AmbientPosition,
+  onComplete: () => void,
+) => {
+  const chainId = useCurrentChain();
+  const { account, signer } = useAccount();
+  const { croc } = useCrocContext();
+  const { poolTokens } = useGetPoolInfo(assetA, assetB);
+  const {
+    minimumPrice: lowerBoundaryPrice,
+    maximumPrice: upperBoundaryPrice,
+    firstAssetValue,
+    secondAssetValue,
+    isFirstAssetOutOfRange,
+    isSecondAssetOutOfRange,
+  } = useDepositContext();
+  const onSubmit = useCallback(async () => {
+    if (!croc || !poolTokens || !signer) {
+      return;
+    }
+
+    const firstAssetBigNumberAmount = isFirstAssetOutOfRange
+      ? BigNumber.from(0)
+      : Decimal.from(firstAssetValue)
+          .asUnits(await poolTokens.tokenA.decimals)
+          .toBigNumber();
+
+    const secondAssetBigNumberAmount = isSecondAssetOutOfRange
+      ? BigNumber.from(0)
+      : Decimal.from(secondAssetValue)
+          .asUnits(await poolTokens.tokenB.decimals)
+          .toBigNumber();
+
+    const transactions: Transaction[] = [];
+
+    const allowanceA = await testAllowance(
+      account,
+      poolTokens.tokenA,
+      firstAssetBigNumberAmount,
+    );
+    const allowanceB = await testAllowance(
+      account,
+      poolTokens.tokenB,
+      secondAssetBigNumberAmount,
+    );
+
+    if (allowanceA) {
+      const approve = await prepareApproveTransaction({
+        token: poolTokens.tokenA.tokenAddr,
+        chain: chainId,
+        amount:
+          allowanceA.weiQty === ethers.constants.MaxUint256
+            ? MaxAllowanceTransferAmount
+            : allowanceA.weiQty,
+        spender: allowanceA.address,
+        contract: new Contract(
+          poolTokens.tokenA.tokenAddr,
+          (
+            await poolTokens.tokenA.context
+          ).erc20Write.interface,
+          signer,
+        ),
+      });
+      if (approve) {
+        transactions.push(approve);
+      }
+    }
+
+    if (allowanceB) {
+      const approve = await prepareApproveTransaction({
+        token: poolTokens.tokenB.tokenAddr,
+        chain: chainId,
+        amount:
+          allowanceB.weiQty === ethers.constants.MaxUint256
+            ? MaxAllowanceTransferAmount
+            : allowanceB.weiQty,
+        spender: allowanceB.address,
+        contract: new Contract(
+          poolTokens.tokenB.tokenAddr,
+          (
+            await poolTokens.tokenB.context
+          ).erc20Write.interface,
+          signer,
+        ),
+      });
+      if (approve) {
+        transactions.push(approve);
+      }
+    }
```

Comment:
The exact same code is in `useHandleSubmit` in `BobDepositModal`. Extract it and unify.